### PR TITLE
Include the scalar tensor auto-transfer in the doc

### LINKED ISSUE #119609: Inconsistent error message about tensors on different devices, when one tensor is a scalar
### 🐛 Describe the bug

I ran into this odd behavior in PyTorch where it _sometimes_ doesn't complain about two tensors being on different devices if one of them is a scalar tensor (_i.e._, has shape `()`). Is this behavior expected? If so, is it documented anywhere?

```python
# two scalars
torch.ones(()) + torch.ones(()).cuda()  # OK
torch.ones(()).cuda() + torch.ones(())  # OK

# one scalar (CPU), one vector (GPU)
torch.ones(()) + torch.ones(1).cuda()  # OK
torch.ones(1).cuda() + torch.ones(())  # OK

# one scalar (GPU), one vector (CPU)
torch.ones(()).cuda() + torch.ones(1)  # fails
torch.ones(1) + torch.ones(()).cuda()  # fails
```

Only the last two lines raise a RuntimeError:

```
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!
```

### Versions

PyTorch 2.1.0

cc @svekars @brycebortree @albanD @malfet

### PR DESCRIPTION
**Pull request by lancerts**
Fixes #119609

@albanD

### CODE REVIEW

File: docs/source/tensor_attributes.rst [hunk 1]
```diff
@@ -227,6 +227,21 @@ non-None device argument.  To globally change the default device, see also
    >>> torch.randn((2,3), device='cuda:1')
    >>> torch.randn((2,3), device=1)  # legacy
 
+.. note::
+   Scalar Tensors (with tensor.dim()==0) are automatically transferred to the GPU device due to the efficiency of this operation. Non-scalar Tensors, however, are not auto-transferred. Additionally, the automatic transfer of Scalar Tensors is uni-directional, only from CPU to GPU, and not vice versa.
```
Comment (albanD) [inline docs/source/tensor_attributes.rst:231]:
```suggestion
   Tensors are never moved automatically between devices and require an explicit call from the user. Scalar Tensors (with tensor.dim()==0) are the only exception to this rule and they are automatically transferred from CPU to GPU when needed as this operation can be done "for free".
```

**Review by albanD — COMMENTED** [review]
Small nit on phrasing

File: docs/source/tensor_attributes.rst (hunk 1 above)
Comment (lancerts) [reply docs/source/tensor_attributes.rst]: Sure, this is much more clear! Thanks for the suggestions.

**Review by albanD — APPROVED** [review]

**Pull request closed without merging**